[BUG] error calling MarshalJSON for type *client.UpdateSamlConfig: bad union

### Describe the bug

```bash
╷
│ Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the
│ plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵

Stack trace from the terraform-provider-vercel_v4.2.0 plugin:

panic: json: error calling MarshalJSON for type *client.UpdateSamlConfig: bad union

goroutine 59 [running]:
github.com/vercel/terraform-provider-vercel/v3/client.mustMarshal(...)
	github.com/vercel/terraform-provider-vercel/v3/client/must_marshal.go:28
github.com/vercel/terraform-provider-vercel/v3/client.(*Client).UpdateTeam(_, {_, _}, {{0x14000156520, 0x1d}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}, ...})
	github.com/vercel/terraform-provider-vercel/v3/client/team.go:188 +0x2e0
github.com/vercel/terraform-provider-vercel/v3/vercel.(*teamConfigResource).Update(0x14000088048, {0x10590acd8, 0x140000e8d20}, {{{{0x105910558, 0x1400021f5f0}, {0x1057cb3a0, 0x140001d73b0}}, {0x1059129f8, 0x140000c60a0}}, {{{0x105910558, ...}, ...}, ...}, ...}, ...)
	github.com/vercel/terraform-provider-vercel/v3/vercel/resource_team_config.go:589 +0x40c
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).UpdateResource(0x140002a8308, {0x10590acd8, 0x140000e8d20}, 0x1400023f3f8, 0x1400023f3c8)
	github.com/hashicorp/terraform-plugin-framework@v1.17.0/internal/fwserver/server_updateresource.go:150 +0x750
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ApplyResourceChange(0x140002a8308, {0x10590acd8, 0x140000e8d20}, 0x14000438230, 0x1400023f638)
	github.com/hashicorp/terraform-plugin-framework@v1.17.0/internal/fwserver/server_applyresourcechange.go:118 +0x140
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ApplyResourceChange(0x140002a8308, {0x10590acd8?, 0x140000e8c30?}, 0x140007123c0)
	github.com/hashicorp/terraform-plugin-framework@v1.17.0/internal/proto6server/server_applyresourcechange.go:71 +0x430
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0x140002a2640, {0x10590acd8?, 0x140000e8030?}, 0x14000428400)
	github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov6/tf6server/server.go:944 +0x294
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0x1058e25a0, 0x140002a2640}, {0x10590acd8, 0x140000e8030}, 0x14000428000, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:789 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x14000380488, {0x10590acd8, 0x140000c8b10}, 0x1400027eb60, 0x140003833e0, 0x105f6aea8, 0x0)
	google.golang.org/grpc@v1.77.0/server.go:1428 +0xd90
google.golang.org/grpc.(*Server).handleStream(0x14000380488, {0x10590b758, 0x14000490000}, 0x1400027eb60)
	google.golang.org/grpc@v1.77.0/server.go:1832 +0xaa8
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.77.0/server.go:1063 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 8
	google.golang.org/grpc@v1.77.0/server.go:1074 +0x138

Error: The terraform-provider-vercel_v4.2.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### To Reproduce

Provide any valid `vercel_team_config` with:

```terraform
  saml = {
    enforced = false
    roles = {
      "eng" = {
        access_group_id = "1"
      }
    }
  }

### Terraform and Vercel Provider Version

```console
2025-12-17T16:40:36.475+0100 [DEBUG] using github.com/hashicorp/go-tfe v1.94.0
2025-12-17T16:40:36.475+0100 [DEBUG] using github.com/hashicorp/hcl/v2 v2.24.0
2025-12-17T16:40:36.475+0100 [DEBUG] using github.com/hashicorp/terraform-svchost v0.1.1
2025-12-17T16:40:36.475+0100 [DEBUG] using github.com/zclconf/go-cty v1.16.3
2025-12-17T16:40:36.475+0100 [INFO]  Go runtime version: go1.25.5
```

### Additional Details

_No response_

--- a/vercel/resource_team_config.go
+++ b/vercel/resource_team_config.go
@@ -229,7 +229,6 @@ type SamlDirectory struct {
 	State types.String `tfsdk:"state"`
 }
 
-// only one of these is non-nil
 type SamlRoles struct {
 	Role          types.String `tfsdk:"role"`
 	AccessGroupID types.String `tfsdk:"access_group_id"`
@@ -310,9 +309,20 @@ func (r *Saml) toUpdateTeamRequest() *client.UpdateSamlConfig {
 	}
 	roles := map[string]client.UpdateSamlConfigRole{}
 	for k, v := range r.Roles {
+		var role *string
+		if !v.Role.IsNull() && !v.Role.IsUnknown() {
+			role = v.Role.ValueStringPointer()
+		}
+		var accessGroupID *string
+		if !v.AccessGroupID.IsNull() && !v.AccessGroupID.IsUnknown() {
+			accessGroupID = v.AccessGroupID.ValueStringPointer()
+		}
+		if role == nil && accessGroupID == nil {
+			continue
+		}
 		roles[k] = client.UpdateSamlConfigRole{
-			Role:          v.Role.ValueStringPointer(),
-			AccessGroupID: v.AccessGroupID.ValueStringPointer(),
+			Role:          role,
+			AccessGroupID: accessGroupID,
 		}
 	}
 	return &client.UpdateSamlConfig{
@@ -389,18 +399,13 @@ func convertResponseToTeamConfig(ctx context.Context, response client.Team, avat
 	if response.Saml != nil {
 		roles := map[string]SamlRoles{}
 		for k, v := range response.Saml.Roles {
-			role := SamlRoles{}
-			if v.Role != nil {
-				role = SamlRoles{
-					Role: types.StringPointerValue(v.Role),
-				}
+			if v.Role == nil && v.AccessGroupID == nil {
+				continue
 			}
-			if v.AccessGroupID != nil {
-				role = SamlRoles{
-					AccessGroupID: types.StringPointerValue(v.AccessGroupID),
-				}
+			roles[k] = SamlRoles{
+				Role:          types.StringPointerValue(v.Role),
+				AccessGroupID: types.StringPointerValue(v.AccessGroupID),
 			}
-			roles[k] = role
 		}
 
 		var diags diag.Diagnostics

